content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md

```diff
@@ -0,0 +1,382 @@
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.6.1</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.javatechie</groupId>
+	<artifactId>springboot-k8s-demo</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>springboot-k8s-demo</name>
+	<description>Demo project for Spring Boot</description>
+	<properties>
+		<java.version>1.8</java.version>
+	</properties>
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+		</plugins>
+		<finalName>springboot-k8s-demo</finalName>
+	</build>
+
+</project>
+
+$﻿ mvn compile
+$﻿ mvn package
+$ ls target/springboot-k8s-demo.jar
+target/springboot-k8s-demo.jar
+```
+
+You can see the *jar* package *springboot-k8s-demo.jar* is built to the *target* folder. B﻿y building a *Docker* image using this generated *jar* file, the *SpringBoot* Web app can be easily deployed as a containerized application to the Kubernetes cluster.
+
+Apache SkyWalking provides a list of agents for instrumenting applications. A specific agent per programming language can be used for building into corresponding service which collects application data and exports them to the SkyWalking OAP server.   
+
+![](/img/sw-agents.png)
+
+I﻿n order to monitor the sample  *SpringBoot* Web app from Apache SkyWalking, we need to download the Java agent and rebuild the image.
+
+![](/img/java-agent.png)
+
+H﻿ere is the *Dockerfile* for building the image with Java agent:
+
+```markdown
+$ cat Dockerfile
+FROM adoptopenjdk:11-jre-hotspot
+# copy extracted java agent folder from the downloaded apache skywalking archive
+ADD agent /opt/agent
+# copy the app jar file
+EXPOSE 8080
+ADD target/springboot-k8s-demo.jar /app/springboot-k8s-demo.jar
+WORKDIR /app
+ENTRYPOINT ["java","-javaagent:/opt/agent/skywalking-agent.jar=agent.namespace=default,agent.service_name=springboot-k8s-app,collector.backend_service=skywalking-oap.skywalking.svc.cluster.local:11800, plugin.jdbc.trace_sql_parameters = true,profile.active=true","-jar","/app/springboot-k8s-app.jar"]
+```
+
+### Monitor SpringBoot application from SkyWalking UI
+
+After build the Docker image *guopingjia/springboot-k8s-demo:pce* and push it to the *DockerHub* registry, we deploy the *SpringBoot* Web app to the Kubernetes cluster with the *deployment.yaml* manifest file:
```

Comment:
Modify first sentence to "After **building** the Docker image *guopingjia/springboot-k8s-demo:pce* and **pushing** it to the *DockerHub* registry, **deploy** the *SpringBoot* **web** app to the Kubernetes cluster with the *deployment.yaml* manifest file: